openedx/core/djangoapps/programs/utils.py

```diff
@@ -151,7 +193,12 @@ def progress(self, programs=None, count_only=True):
                 if self._is_course_complete(course):
                     completed.append(course)
                 elif self._is_course_in_progress(course):
-                    in_progress.append(course)
+                    course_in_progress = self.does_course_still_qualify_as_in_progress(now, enrolled_run_modes, course)
+                    if course_in_progress:
+                        in_progress.append(course)
+                    else:
+                        course['expired'] = True
```

Comment:
it would add the flag there, but it is already creating new data?  
I did originally design this with a separate expired section in the progress bar, but i realized that doesn't really make sense.  
for the purposes of choosing a section it belongs in in progress but the flag tells us to change the display  
did you have a suggestion?